src/redux/teacher/teacherReducer.js

```diff
@@ -0,0 +1,32 @@
+import { FETCH_TEACHERS_FAILURE, FETCH_TEACHERS_REQUEST, FETCH_TEACHERS_SUCCESS } from "./teacherTypes"
+
+const initialState = {
+    loading: false,
+    teachers: [],
+    error: ''
+}
+
+const reducer = (state = initialState, action) => {
```

Comment:
as it's about teacher you can call it so. 
`const teachers = (state = initialState, action) =>{ ....`